File: backend/ClientApp/src/types/ticket.ts
export enum TicketStatus {
  Open     = 'Open',
  Resolved = 'Resolved',
  Closed   = 'Closed',
}

export enum TicketCategory {
  GeneralQuestion   = 'GeneralQuestion',
  TechnicalQuestion = 'TechnicalQuestion',
  RefundRequest     = 'RefundRequest',
}

export interface Ticket {
  id: number;
  senderEmail: string;
  subject: string;
  status: TicketStatus;
  category: TicketCategory | null;
  assignedTo: string | null;
  createdAt: string;
}

export interface TicketDetail extends Ticket {
  body: string;
  aiSummary: string | null;
  aiSuggestedReply: string | null;
  resolvedAt: string | null;
}

export interface TicketsPage {
  items: Ticket[];
  totalCount: number;
  page: number;
  pageSize: number;
}

File: backend/Models/Ticket.cs
namespace HelpdeskAi.Models;

public class Ticket
{
    public int Id { get; set; }
    public required string SenderEmail { get; set; }
    public required string Subject { get; set; }
    public required string Body { get; set; }
    public string? MessageId { get; set; }
    public TicketStatus Status { get; set; } = TicketStatus.Open;
    public TicketCategory? Category { get; set; }
    public string? AssignedToId { get; set; }
    public ApplicationUser? AssignedTo { get; set; }
    public string? AiSummary { get; set; }
    public string? AiSuggestedReply { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? ResolvedAt { get; set; }
}

public enum TicketStatus { Open, Resolved, Closed }

public enum TicketCategory { GeneralQuestion, TechnicalQuestion, RefundRequest }
